platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/crypto/EnhancedKeyStoreLoader.java

```diff
@@ -309,6 +330,29 @@ public EnhancedKeyStoreLoader scan() throws KeyLoadingException, KeyStoreExcepti
         return this;
     }
 
+    @Nullable
+    private BlsPrivateKey resolveTssPrivateKey(@NonNull final NodeId nodeId, @NonNull final String nodeAlias)
+            throws KeyLoadingException {
+        Objects.requireNonNull(nodeId, MSG_NODE_ID_NON_NULL);
+        Objects.requireNonNull(nodeAlias, MSG_NODE_ALIAS_NON_NULL);
+
+        Path keyLocation = keyStoreDirectory.resolve(String.format("t-private-%s.tss", nodeAlias));
+        if (Files.exists(keyLocation)) {
+            logger.trace(
+                    STARTUP.getMarker(),
+                    "Found tss encryption private key for node {} [ fileName = {} ]",
+                    nodeId,
+                    keyLocation.getFileName());
+            try {
+                return AsciiArmoredFiles.readPrivateKey(keyLocation);
+            } catch (final Exception e) {
+                logger.error(ERROR.getMarker(), "Failed to read TSS encryption private key from disk", e);
+                throw new KeyLoadingException("Failed to read TSS encryption private key from disk", e);
```

Comment:
Yes we could return null and let the private key be regenerated.

However, this behavior should be clear in the node operator documentation. In the case of the private key becoming corrupted on-disk, it would be regenerated and a new public key submitted without giving the operator a chance to restore the private key from a backup.

Reply:
Updated to return null, key will be regenerated.